Bio/Phylo/MCMC.py

```diff
@@ -0,0 +1,530 @@
+# Copyright (C) 2020 by Magda Grynkiewicz (magda.markowska@gmail.com)
+
+"""Classes and methods for MCMC sampling procedure."""
+
+import random
+import math
+import matplotlib as plt
+import copy
+from Bio import Phylo
+from Bio.Phylo.TreeConstruction import DistanceCalculator
+from Bio.Phylo.TreeConstruction import DistanceTreeConstructor
+from Bio.Phylo.TreeConstruction import LikelihoodScorer
+from Bio import AlignIO
+from Bio.Align import MultipleSeqAlignment
+from Bio.Phylo.EvolutionModel import GTRModel
+from Bio.Phylo.EvolutionModel import EvolutionModel
+from Bio.Phylo.EvolutionModel import F81Model
+
+
+class Stepper:
+    """Base class for one simulation step.
+
+    :Parameters:
+        size_param: float
+            Step size parameter sometimes called tuning parameter.
+            Should be positive.
+    """
+
+    def __init__(self, size_param=False):
+        """Init method for Stepper."""
+        if not size_param:
+            raise ValueError("!")
+        else:
+            self._size_param = self._validate_size_param(size_param)
```

Comment:
Ale tutaj to właściwie nie jest potrzebne chyba, po prostu możesz zostawić parametr size_param bez defaultowej wartości i i tak wywali error jeśli go nie dasz. Można wtedy zostawić tylko 34 linijkę